scene/gui/split_container.cpp

```diff
@@ -153,109 +146,119 @@ Ref<Texture2D> SplitContainer::_get_grabber_icon() const {
 	Control *second = _get_sortable_child(1);
+	int axis_index = vertical ? 1 : 0;
+	int size = get_size()[axis_index];
+	int sep = _get_separation();
 
-	// Determine expanded children.
-	bool first_expanded = (vertical ? first->get_v_size_flags() : first->get_h_size_flags()) & SIZE_EXPAND;
-	bool second_expanded = (vertical ? second->get_v_size_flags() : second->get_h_size_flags()) & SIZE_EXPAND;
-
-	// Compute the minimum size.
-	int axis = vertical ? 1 : 0;
-	int size = get_size()[axis];
-	int ms_first = first->get_combined_minimum_size()[axis];
-	int ms_second = second->get_combined_minimum_size()[axis];
-
-	// Determine the separation between items.
-	Ref<Texture2D> g = _get_grabber_icon();
-	int sep = (dragger_visibility != DRAGGER_HIDDEN_COLLAPSED) ? MAX(theme_cache.separation, vertical ? g->get_height() : g->get_width()) : 0;
-
-	// Compute the wished separation_point.
-	int wished_middle_sep = 0;
+	// Compute the wished size.
+	int wished_size = 0;
 	int split_offset_with_collapse = 0;
 	if (!collapsed) {
 		split_offset_with_collapse = split_offset;
 	}
-	if (first_expanded && second_expanded) {
+	bool first_is_expanded = (vertical ? first->get_v_size_flags() : first->get_h_size_flags()) & SIZE_EXPAND;
+	bool second_is_expanded = (vertical ? second->get_v_size_flags() : second->get_h_size_flags()) & SIZE_EXPAND;
+
+	if (first_is_expanded && second_is_expanded) {
 		float ratio = first->get_stretch_ratio() / (first->get_stretch_ratio() + second->get_stretch_ratio());
-		wished_middle_sep = size * ratio - sep / 2 + split_offset_with_collapse;
-	} else if (first_expanded) {
-		wished_middle_sep = size - sep + split_offset_with_collapse;
+		wished_size = size * ratio - sep * 0.5 + split_offset_with_collapse;
+	} else if (first_is_expanded) {
+		wished_size = size - sep + split_offset_with_collapse;
 	} else {
-		wished_middle_sep = split_offset_with_collapse;
+		wished_size = split_offset_with_collapse;
 	}
 
-	// Clamp the middle sep to acceptatble values.
-	middle_sep = CLAMP(wished_middle_sep, ms_first, size - sep - ms_second);
+	// Clamp the split offset to acceptable values.
+	int first_min_size = first->get_combined_minimum_size()[axis_index];
+	int second_min_size = second->get_combined_minimum_size()[axis_index];
+	computed_split_offset = CLAMP(wished_size, first_min_size, size - sep - second_min_size);
 
 	// Clamp the split_offset if requested.
 	if (p_clamp) {
-		split_offset -= wished_middle_sep - middle_sep;
+		split_offset -= wished_size - computed_split_offset;
 	}
 }
 
 void SplitContainer::_resort() {
 	Control *first = _get_sortable_child(0);
 	Control *second = _get_sortable_child(1);
 
-	// If we have only one element.
-	if (!first || !second) {
+	if (!first || !second) { // Only one child.
 		if (first) {
 			fit_child_in_rect(first, Rect2(Point2(), get_size()));
 		} else if (second) {
 			fit_child_in_rect(second, Rect2(Point2(), get_size()));
 		}
 		dragging_area_control->hide();
 		return;
+	} else if (!collapsed) {
```

Comment:
```suggestion
	}
	if (!collapsed) {
```
This can be separate.